s4e-backend/src/main/java/pl/cyfronet/s4e/service/ReportTemplateService.java

```diff
@@ -0,0 +1,65 @@
+package pl.cyfronet.s4e.service;
+
+import lombok.Builder;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import lombok.val;
+import org.springframework.data.domain.Sort;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.cyfronet.s4e.data.repository.ReportTemplateRepository;
+import pl.cyfronet.s4e.ex.NotFoundException;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class ReportTemplateService {
+    @Value
+    @Builder
+    public static class CreateDTO {
+        String ownerEmail;
+        String caption;
+        String notes;
+        List<Long> overlayIds;
+        Long productId;
+    }
+
+    private final ReportTemplateRepository reportTemplateRepository;
+    private final ReportTemplateMapper reportTemplateMapper;
+
+    @Transactional(rollbackFor = NotFoundException.class)
+    public UUID create(CreateDTO createDTO) throws NotFoundException {
```

Comment:
Wouldn't we need some additional validation for products and overlays ids?

Reply:
It's not crucial to keep these references correct if I understand the functionality correctly, so I don't validate them.